framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -934,6 +943,11 @@ public Boolean visitWildcard_Intersection(
     @Override
     public Boolean visitWildcard_Primitive(
             AnnotatedWildcardType subtype, AnnotatedPrimitiveType supertype, VisitHistory visited) {
+        if (subtype.isUninferredTypeArgument()) {
```

Comment:
If the wildcards is `? extends Integer` and super is `int`, then visitWildcardSubtype will call isSubtype on `Integer` and `int`.  That call will unbox `Integer`.